### ISSUE TITLE
Traing process is killed due to very high ram consumption

### ISSUE THREAD

**Issue by songhc8** [roboflow/rf-detr#820]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

Dataset: ~3000 images in YOLO format about 18GB. Resolution is varying and the largest width or height is leq 1024. 
Hardware: RAM 64GB, RTX5090D-32G,
Training Config:`# from rfdetr import RFDETRSegMedium

traning log:
```
[2026-03-19 06:27:26] [WARNING] rf-detr - Using patch size 12 instead of 14, which means we're not loading DINOv2 backbone weights. This is not a problem if finetuning a pretrained RF-DETR model.

[2026-03-19 06:27:27] [INFO] rf-detr - File /song/work/tmp/python_tmp/rf-detr/rfdetr_cpp/rf-detr-seg-nano.pt already exists with correct MD5 hash.

[2026-03-19 06:27:28] [WARNING] rf-detr - Using a different number of positional encodings than DINOv2, which means we're not loading DINOv2 backbone weights. This is not a problem if finetuning a pretrained RF-DETR model.

[2026-03-19 06:27:28] [WARNING] rf-detr - Using patch size 12 instead of 14, which means we're not loading DINOv2 backbone weights. This is not a problem if finetuning a pretrained RF-DETR model.

[2026-03-19 06:27:30] [INFO] rf-detr - File /song/work/tmp/python_tmp/rf-detr/rfdetr_cpp/rf-detr-seg-nano.pt already exists with correct MD5 hash.

[2026-03-19 06:27:30] [WARNING] rf-detr - TensorBoard logging disabled: Neither `tensorboard` nor `tensorboardX` is available. Try `pip install`ing either.

Requirement 'tensorboardX' not met. HINT: Try running `pip install -U 'tensorboardX'`

Requirement 'tensorboard' not met. HINT: Try running `pip install -U 'tensorboard'`. Install with: pip install tensorboard
Using bfloat16 Automatic Mixed Precision (AMP)

GPU available: True (cuda), used: True

TPU available: False, using: 0 TPU cores

💡 Tip: For seamless cloud logging and experiment tracking, try installing [litlogger](https://pypi.org/project/litlogger/) to enable LitLogger, which logs metrics and artifacts automatically to the Lightning Experiments platform.

[2026-03-19 06:27:30] [INFO] rf-detr - Using multi-scale training with square resize and scales: [372]

[2026-03-19 06:27:30] [INFO] rf-detr - Built 1 Albumentations transforms from config

[2026-03-19 06:27:30] [INFO] rf-detr - Built 1 Albumentations transforms from config

Killed
```
When training is start, all Ram is used and after that the training process is being killed by system. 

### Environment
```
RD-DETR 1.6.0.rc1
Ubuntu24.04
python 3.12
torch 2.10.0
CUDA/CUDNN 12.8/9.10.2.21
GPU RTX5090D-32GB
```
### Minimal Reproducible Example
```py
from rfdetr import RFDETRSegNano
from rfdetr.datasets.aug_config import AUG_CONSERVATIVE


model = RFDETRSegNano(pretrain_weights="rf-detr-seg-nano.pt")

data_name="carton"

model.train(
    dataset_dir=f"./data/{data_name}",
    epochs=50,
    batch_size=4,
    device = "cuda",
    grad_accum_steps=2,
    num_workers=2,
    lr=1e-4,
    output_dir=f"./runs_{data_name}",
    aug_config=AUG_CONSERVATIVE,
    fp16_eval=True,
    accelerator="gpu",
    eval_max_dets=50,
    eval_interval=2,
    early_stopping=True,
    early_stopping_patience=5,
)`
```

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!

**Comment by SebastianJanampa** [comment]
How many objects per image does your dataset have?
A reason for the high memory consumption could be the use of group queries.  So you could play with the grad_accum_steps and the batch_size variables

**Comment by songhc8** [comment]
<img width="1216" height="989" alt="Image" src="https://github.com/user-attachments/assets/fa800819-4974-4c7a-9b9b-e2a3e6f03562" />

I reinstall the project with latest version. But the bus still exists.  here is my tring script:
```
# from rfdetr import RFDETRSegMedium
from rfdetr import RFDETRSegNano
from rfdetr.datasets.aug_config import AUG_AGGRESSIVE,AUG_CONSERVATIVE


model = RFDETRSegNano(pretrain_weights="rf-detr-seg-nano.pt")

data_name="carton"

model.train(
    dataset_dir=f"./data/{data_name}",
    epochs=50,
    batch_size=4,
    device = "cuda",
    # use_ema=True,
    grad_accum_steps=2,
    num_workers=8,
    pin_memory=True,
    persistent_workers=True,
    prefetch_factor=4,
    output_dir=f"./runs_{data_name}_3",
    aug_config=AUG_CONSERVATIVE,
    accelerator="auto",
    early_stopping=True,
    early_stopping_patience=5,
    # early_stopping_use_ema=True
)
```

Dataset is in yolo format. There are about 6000 images in training set,28GB.

**Comment by Borda** [comment]
@songhc8 can you please tell us more about the dataset, you mentioned image size 1024, but how many classes, typical number of instances per image? Ideally could you share the dataset so we can reproduce it?

**Comment by songhc8** [comment]
> [@songhc8](https://github.com/songhc8) can you please tell us more about the dataset, you mentioned image size 1024, but how many classes, typical number of instances per image? Ideally could you share the dataset so we can reproduce it?

@Borda Only 1 class, ~6000 images in YOLO format. the number of instances per image is varying, maybe from 4 to 20. Very sorry for that I can not share the dataset, due to the policy of company.

**Comment by songhc8** [comment]
@Borda I notice one thing that the class label is start with 0, not 1 in yolo annotation txt file. Does it matter? When only 1 class exists in my dataset.
